src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,39 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 200) {
+    this.health = health;
+    this.name = 'Animal';
+  }
+
+  dead(healthPoints) {
+    this.health += healthPoints;
+
+    if (this.health < 100) {
+      Animal.alive = Animal.alive.filter((animal) => animal.health);
+    }
```

Comment:
[LOGIC ERROR] The condition should check if health is less than or equal to 0 to determine if the animal is dead. Also, the filter condition is incorrect; it should remove the dead animal, not keep animals with any health value.